### DISCUSSION TITLE
SkyWalking Java Agent is being host in skywalking-java repo since 8.8.0

### DISCUSSION THREAD

**Discussion by wu-sheng** [apache/skywalking#7548] [General]
# Update
The new Java Agent repository, **https://github.com/apache/skywalking-java**, has been created. 
**The source codes of agent core and plugins are all hosted there.**

# Background
In case not everyone noticed, the main CI-IT task time of our main repo is closing in one hour.
Our e2e tasks and plugin tests are still efficient, but CI-IT will be a time blocker for us shortly.

In my mind, the only option would be to split the java agent from the main repo codebase and create a skywalking-java repo instead. This is still not urgent, but the trend is very clear. When we reach that point, we will not have any so-called main-repo anymore. Everything would be a component, and replaceable.

There is no action planned to be taken, but I hope everyone could know this day will come eventually. If the time is over 60 mins, we may have no option, but to do it.

Mail list link, https://lists.apache.org/thread.html/r9a4950cfd0240447d73fad56b98beebf7fb4d9718a7e65b786f786f9%40%3Cdev.skywalking.apache.org%3E

**Comment by wu-sheng** [comment]
We could consider to do this in 8.9.0, if more supports show up.

**Reply by wu-sheng** [reply]
Now, we don't have agent related plugin, we are going to do it now, and include it in 8.8.0 release directly.

**Comment by zlay0701** [comment]
supports

**Comment by dao-jun** [comment]
Are there any module referred by both of them(the java-agent and backend)?  How to deal with it if there was.

**Reply by wu-sheng** [reply]
There are some, which is why this has been be certain. We have to release them in a separate repo by using new version number, such as 9.0.0. Or make them duplicate in 2 different repo(s).

**Reply by wu-sheng** [reply]
FYI @apache/skywalking-committers We need to consider this for the next move.

**Reply by wu-sheng** [reply]
According to @kezhenxu94 's https://github.com/apache/skywalking/issues/7553#issuecomment-904646562, we are going to be tolerant with duplicate codes(necessary codes) at both repo(s), mostly DataCarrier..

**Comment by Superskyyy** [comment]
I totally support the split, when I first met SkyWalking few months ago, it did took me some time thinking why Java agent is bundled with the OAP while the others aren't.

It is also bit tricker to understand the codebase for newcomers just to distinguish which part belongs to the backend, which goes to the agent. I believe splitting would make everything more clear and attract even more developers to contribute.

**Reply by kezhenxu94** [reply]
When I mentor students in summer of code activities and other new contributors, they share exactly the same situation as yours.

**Reply by wu-sheng** [reply]
Thanks for the feedback. 

An update for everyone reading this. SkyWalking was a very small project for fun and training only. So, we thought we will have a runnable solution for Java ecosystem only, even everything goes well.

Now, here we come.

**Reply by DearZh** [reply]
> When I mentor students in summer of code activities and other new contributors, they share exactly the same situation as yours.

I also want to be a new contributor.  How can I get your guidance? 
 
Whether there is a corresponding communication group?  such as WeChat？ 
 
I hope I can contribute to skywalking. thank you

**Reply by Superskyyy** [reply]
> > When I mentor students in summer of code activities and other new contributors, they share exactly the same situation as yours.
> 
> I also want to be a new contributor. How can I get your guidance?
> 
> Whether there is a corresponding communication group? such as WeChat？
> 
> I hope I can contribute to skywalking. thank you

Hello @DearZh! Thank you for your interest, it is the way how open-source at Apache Foundation works when we all recommend communications to be open and public while also being asynchronous. Though there are Slack and QQ chat groups available, you can find them
at https://github.com/apache/skywalking#contact-us here. 

Other than that, is there any particular problem that blocks you from starting right now (things like do not know which issues to work on, understanding the project overall etc.) ? 

Try open a new discussion thread and we can proceed there to help you start your open-source journey.  BTW, if you are a current student, you can expect the Summer 2022 open-source event opening for application in 1 month, you can try applying for a project and you will indeed get a mentor over the next few months :)

**Reply by DearZh** [reply]
> > > When I mentor students in summer of code activities and other new contributors, they share exactly the same situation as yours.
> > 
> > 
> > I also want to be a new contributor. How can I get your guidance?
> > Whether there is a corresponding communication group? such as WeChat？
> > I hope I can contribute to skywalking. thank you
> 
> Hello @DearZh! Thank you for your interest, it is the way how open-source at Apache Foundation works when we all recommend communications to be open and public while also being asynchronous. Though there are Slack and QQ chat groups available, you can find them at https://github.com/apache/skywalking#contact-us here.
> 
> Other than that, is there any particular problem that blocks you from starting right now (things like do not know which issues to work on, understanding the project overall etc.) ?
> 
> Try open a new discussion thread and we can proceed there to help you start your open-source journey. BTW, if you are a current student, you can expect the Summer 2022 open-source event opening for application in 1 month, you can try applying for a project and you will indeed get a mentor over the next few months :)

Cool, everyone's reply is full of goodwill. I don't have any special problems for the time being,I'm in the familiar stage. But thank you for your reply,thanks you. @Superskyyy

**Comment by JaredTan95** [comment]
make sense, +1 for me.

**Comment by wu-sheng** [comment]
Repo created, https://github.com/apache/skywalking-java

**Reply by ruoshuixuelabi** [reply]
where is java-agent 8.8.0 version？the latest is 8.7.0

**Reply by kezhenxu94** [reply]
> where is java-agent 8.8.0 version？the latest is 8.7.0

@ruoshuixuelabi it's not released yet, after 8.7.0 the java agent will be released separately

**Reply by ruoshuixuelabi** [reply]
in other words ，SkyWalking 8.8.0 is released ，but  java agent  8.8.0 not released 。i want use SkyWalking 8.8.0 It's impossible？

**Reply by kezhenxu94** [reply]
@ruoshuixuelabi yes, you can use OAP 8.8.0 with java agent 8.7.0, please check the [compatibility doc](https://skywalking.apache.org/docs/main/v8.8.0/en/setup/service-agent/agent-compatibility/)

**Reply by wu-sheng** [reply]
We created compatibility document for everyone already.

https://skywalking.apache.org/docs/main/latest/en/setup/service-agent/agent-compatibility/

I don't know what do you mean impossible, Java agent is an agent, nothing more.

**Reply by ruoshuixuelabi** [reply]
in other words，i can use SkyWalking 8.8.0  with java agent 8.7.0？If so, I understand

**Reply by ruoshuixuelabi** [reply]
thanks

**Comment by zlay0701** [comment]
https://github.com/apache/skywalking-java



zlay  ***@***.***
nothing's gonna change my love for you
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 
From: ruoshuixuelabi
Date: 2021-09-29 14:08
To: apache/skywalking
CC: 素喂; Comment
Subject: Re: [apache/skywalking] SkyWalking Java Agent is being host in skywalking-java repo since 8.8.0 (Discussion #7548)
where is java-agent 8.8.0 version？the latest is 8.7.0
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by heihei180** [comment]
Nice, the project structure is clearer

**Comment by dmsolr** [comment]
Cool, my MacBook is going to calm down.
